### ISSUE TITLE
Propability of data augmentation techniques

### ISSUE THREAD

**Issue by eypros** [ultralytics/ultralytics#18613]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am using Yolov8 for my project and I was interested in the augmentation being applied during the training session. There are certain questions I would like to clarify regarding the augmentation options.

As mentioned in this [issue](https://github.com/ultralytics/ultralytics/issues/2412) I can pass the augmentation parameters in the conf yaml file used for training.

But as I understand it I can also pass them as arguments in the `model.train` call as in [here](https://github.com/ultralytics/ultralytics/issues/8308). 

1) I guess both options are valid. In this case which one has priority over the other (if an option affects the same technique). Also, can they mix up without problem (specifying some in the model.train and some in the yaml file should consider both techniques).

2) In this [issue](https://github.com/ultralytics/ultralytics/issues/2412) it is mentioned that some augmentation techniques take probability as argument (as `mosaic` for example). This is clear for me. Regarding the ones that take other values, as for example `degrees` which expects DEGREES for rotation. So, as I understand it in the later case this means the rotation is applied ALWAYS (100% of the time) but the rotation angle varies from -DEGREE to +DEGREES (e.g. from -30 to 30 degrees). Is this the case?

3) Since I consider trying also Yolov11 the general idea for applying augmentation techniques remains the same (through parameters etc.)?

Thank you

**Comment by UltralyticsAssistant** [comment]
👋 Hello @eypros, thank you for your interest in Ultralytics 🚀! We recommend exploring the [Ultralytics Documentation](https://docs.ultralytics.com) for detailed explanations, including augmentation options and practical examples using [Python](https://docs.ultralytics.com/usage/python/) or [CLI](https://docs.ultralytics.com/usage/cli/).

If this is a ❓ Question about custom training, augmentations, or related topics, please provide more specifics if possible. These could include snippets of your `.yaml` configuration, `model.train()` arguments, or example workflows to help clarify your interpretation. For augmentation-specific details, you may find our [Model Training Tips Guide](https://docs.ultralytics.com/guides/model-training-tips/) helpful.

For 🐛 Bug Reports, kindly include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better understand the scenario you're encountering.

---

### Augmentation

Specific to your augmentation-related queries:
1. To determine the priority between augmentations specified in `.yaml` and those passed directly via `model.train()`, further clarification might be required. Feel free to test combinations and share any inconsistencies you might observe, as your insights could strengthen best practices for users.
2. You seem to understand correctly regarding probabilistic (e.g., `mosaic`) vs. value-based augmentations (e.g., `degrees`), but if you suspect a misunderstanding in behavior, we encourage you to isolate questionable augmentations and test them to validate their effects during training. Sharing these findings could help the community!
3. Regarding YOLOv11, future versions will likely carry forward key ideas but may involve slight variations. Following the [Changelog](https://github.com/ultralytics/ultralytics/releases) would keep you updated with any modifications introduced.

---

### Upgrade

Ensure you are using the latest version of the `ultralytics` package so your questions and observations align with the most updated features and bug fixes. Upgrade it with:

```bash
pip install -U ultralytics
```

---

### Environments

YOLO can be seamlessly executed in the following environments for optimal performance:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

### Community Support

Join the Ultralytics community to engage with experts and enthusiasts:
- For quick real-time discussions, try [Discord](https://discord.com/invite/ultralytics) 🎧.
- Have in-depth technical queries or new ideas? Visit [Discourse](https://community.ultralytics.com).
- Interested in community-driven discussions? Check out our [Subreddit](https://reddit.com/r/Ultralytics).

---

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

The CI status badge reflects the health of all [Ultralytics CI tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) which verify the latest developments on macOS, Windows, and Ubuntu across tasks and modes.

---

⚠️ This is an automated response to help expedite assistance. You can expect an Ultralytics engineer to follow up with you soon! Let us know if you uncover any additional insights in the meantime. 😊

**Comment by Y-T-G** [comment]
1. The manually passed arguments take priority. https://github.com/ultralytics/ultralytics/blob/cc1e77138c2a124563b898d8e267a3045fa7bedf/ultralytics/engine/model.py#L796
2. `degrees` would randomly select a degree between `±degrees`. So if it's not applied if it's randomly 0.
3. They should be same
